dali/dfuXRefLib/dfuxreflib.cpp

```diff
@@ -994,17 +1010,17 @@ static bool parseFileName(const char *name,StringBuffer &mname,unsigned &num,uns
 
             // Check for dir-per-part number
             const char *tailSlash = cur;
-            while (*tailSlash&&*tailSlash!='/')
+            while (tailSlash!=name&&*tailSlash!='/')
                 tailSlash--;
             const char *d = tailSlash-1;
-            for (int i=1;*(d)&&isdigit(*d);i*=10,d--)
+            for (int i=1;d!=name&&isdigit(*d);i*=10,d--)
                 dirPerPart += (*d-'0')*i;
+            // If a dir-per-part number was found, check that it matches the part number
             if (*d&&*d=='/')
```

Comment:
similar to a prior comment - I don't think it is valid here to be testing *d for '\0' - that should never be possible. If it null, it suggests it has walked past the beginning of the input string and into memory where it has hit a 0.